llvm/lib/Target/AMDGPU/AMDGPUPromoteAlloca.cpp

```diff
@@ -920,6 +919,12 @@ bool AMDGPUPromoteAllocaImpl::tryPromoteAllocaToVector(AllocaInst &Alloca) {
       Value *Index = GEPToVectorIndex(GEP, &Alloca, VecEltTy, *DL, NewGEPInsts);
       if (!Index)
         return RejectUser(Inst, "cannot compute vector index for GEP");
+      // Alternatively, if there is a constant index for which we already know
+      // that it will be out-of-bounds, we also don't want to promote this
+      // alloca to vector.
+      if (ConstantInt *I = dyn_cast<ConstantInt>(Index);
+          I && I->getZExtValue() >= Alloca.getAllocationSize(*DL))
+        return RejectUser(Inst, "GEP constant index out-of-bounds");
```

Comment:
There's no reason to reject it, and you should proceed. The out of bounds index will just produce a poison value in the vector indexing

Reply:
I see, I think I now get what you mean